R/modules.R

```diff
@@ -264,7 +264,7 @@ module <- function(label = "module",
   }
   checkmate::assert_list(transformers, types = "teal_transform_module")
   transformer_datanames <- unlist(lapply(transformers, attr, "datanames"))
-  combined_datanames <- if (identical(datanames, "all") || identical(transformer_datanames, "all")) {
+  combined_datanames <- if (identical(datanames, "all")) {
```

Comment:
Possible second scenario and `module()$datanames = "all"` if no custom datanames were provided in both `module(datanames = ` and `transformers(datanames`

```suggestion
  combined_datanames <- if (identical(datanames, "all") ||  identical(unique(transformer_datanames), "all")) {
```

Reply:
```suggestion
  combined_datanames <- if (identical(datanames, "all") || any(sapply(transformers, identical, "all"))) {
```

Reply:
Ok, based on the fact that transformers is a list where it can have multiple datanames I will go with 

`any(sapply(transformer_datanames, identical, "all"))` 

https://github.com/insightsengineering/teal/pull/1344/commits/93b9cf4be69631b979e62efc599df3926887b642